osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisServiceImpl.java

```diff
@@ -824,7 +886,45 @@ public PageOfUsers listUsers(String tenantId, long offset, long limit) {
 
     @Override
     public OsisUser updateUser(String tenantId, String userId, OsisUser osisUser) {
-        throw new NotImplementedException();
+        try {
+            OsisTenant tenant = ScalityModelConverter.toOsisTenant(vaultAdmin.getAccount(ScalityModelConverter.toGetAccountRequestWithID(tenantId)));
+            logger.info("Update User request received:: tenant ID:{}, user ID:{}", tenantId, userId);
+
+            Credentials tempCredentials = getCredentials(tenantId);
+            final AmazonIdentityManagement iam = vaultAdmin.getIAMClient(tempCredentials, appEnv.getRegionInfo().get(0));
+
+            // List all access keys for the user
+            ListAccessKeysRequest listAccessKeysRequest =  ScalityModelConverter.toIAMListAccessKeysRequest(userId, DEFAULT_MAX_LIMIT);
+
+            logger.debug("[Vault] List Access Keys Request:{}", new Gson().toJson(listAccessKeysRequest));
+
+            ListAccessKeysResult listAccessKeysResult = iam.listAccessKeys(listAccessKeysRequest);
+
+            logger.debug("[Vault] List Access Keys response:{}", new Gson().toJson(listAccessKeysResult));
+
+            for(AccessKeyMetadata accessKey: listAccessKeysResult.getAccessKeyMetadata()) {
+                // Update each access key of the user to active/inactive
+                UpdateAccessKeyRequest updateAccessKeyRequest =  ScalityModelConverter.toIAMUpdateAccessKeyRequest(userId,
+                        accessKey.getAccessKeyId(), osisUser.getActive());
+                iam.updateAccessKey(updateAccessKeyRequest);
+            }
+
+            logger.info("Updated user response:{}", ScalityModelConverter.maskSecretKey(new Gson().toJson(osisUser)));
+            return  osisUser;
+        } catch (Exception e){
+
+            if(isAdminPolicyError(e)) {
+                try {
+                    generateAdminPolicy(tenantId);
+                    return updateUser(tenantId, userId, osisUser);
```

Comment:
Is the number of times this can fail bounded in any way (meaning will the error be different after calling `generateAdminPolicy`)? If not, then some throttling/backoff would be needed to avoid DoS-ing ourselves.

Reply:
@RachedBenMustapha  This is not retry for any other exceptions. This retry exists for majority of OSIS APIs where an Admin policy is required and if that is not exists, we recreate the policy and retry the API.

For more info refer: https://github.com/scality/osis/blob/development/1.0/Design.md#setupassumerole-subroutine